### PR TITLE
Add cylinder, move geometry into a file

### PR DESCRIPTION
**Pull request by davepagurek**
Fixes https://github.com/calder-gl/calder/issues/70 and https://github.com/calder-gl/calder/issues/65

It looks like this now:

screen shot 2018-06-13 at 2 39 26 pm

### CODE REVIEW

**Comment by armcburney** [comment]
😍

File: src/examples/generation.ts [hunk 1]
```diff
@@ -33,13 +33,13 @@ renderer.addLight(light2);
 
 // Setup leaf
 const leafColor = RGBColor.fromRGB(204, 255, 204);
-const workingLeafSphere = genSphere(leafColor);
+const workingLeafSphere = Shape.sphere(leafColor);
```
Comment (armcburney) [inline src/examples/generation.ts:36]: 😍

File: src/geometry/Shape.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
+import { vec3 } from 'gl-matrix';
+import { range } from 'lodash';
+
+import { Face, RGBColor, WorkingGeometry } from '../calder';
+import { Color } from '../colors/Color';
+import { genIsoSurface } from './MarchingCubes';
+import { ScalarField } from './ScalarField';
+
+export namespace Shape {
+    export function sphere(fillColor: Color = RGBColor.fromHex('#EEEEEE')): WorkingGeometry {
```
Comment (armcburney) [inline src/geometry/Shape.ts:10]: I like `#EEEEEE` as a default colour 👍

File: src/geometry/Shape.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    export function cylinder(fillColor: Color = RGBColor.fromHex('#EEEEEE')): WorkingGeometry {
+        const LENGTH = 1;
+        const PRECISION = 20;
+
+        const vertices: vec3[] = [];
+        const normals: vec3[] = [];
+
+        // We need to duplicate the vertices so that ones for the faces on the ends can have
+        // different normals from the ones in the middle segment of the cylinder
+        [true, false].forEach((onEnd: boolean) => {
```
Comment (armcburney) [inline src/geometry/Shape.ts:39]: this is genius 💯

File: src/geometry/Shape.ts [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+            });
+        });
+
+        const faces: Face[] = [];
+
+        // Top and bottom faces: connect outer ring vertices to central vertex
+        [0, 1].forEach((side: number) => {
+            range(PRECISION).forEach((i: number) => {
+                const offset = side * (PRECISION + 1);
+                faces.push(new Face([offset, offset + i + 1, offset + (i + 1) % PRECISION + 1]));
```
Comment (armcburney) [inline src/geometry/Shape.ts]: A little bit more documentation here would be nicer. 🙏

File: src/geometry/Shape.ts [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+            });
+        });
+
+        // Connect top and bottom rings
+        range(PRECISION).forEach((i: number) => {
+            const offset = (PRECISION + 1) * 2;
+            faces.push(
+                new Face([offset + i, offset + (i + 1) % PRECISION, offset + PRECISION + i])
```
Comment (armcburney) [inline src/geometry/Shape.ts]: Ditto

**Review by armcburney — APPROVED** [review]
Just a few nits, but other than that - LGTM! 💯I think this makes us good-to-go for the midterm demo!

**Comment by davepagurek** [comment]
Added some more ascii art

**Comment by armcburney** [comment]
Bless, thank you dave

**Review by armcburney — APPROVED** [review]
Nice art lmao

**Pull request merged**